### ISSUE TITLE
SlevomatCodingStandard.TypeHints.ParameterTypeHintSpacing interprets parameter attributes as type

### ISSUE THREAD

**Issue by TimWolla** [slevomat/coding-standard#1476]
Consider the following example:

```php
<?php

function foo(
    #[A]
    $a
) {
    return $a;
}
```

Imagine the code does not have a parameter type for backwards compatibility / interface compatibility. Now running:

```
$ phpcs --standard=SlevomatCodingStandard --sniffs=SlevomatCodingStandard.TypeHints.ParameterTypeHintSpacing test.php
```

Reports:

```
FILE: /pwd/test.php
-----------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------
 4 | ERROR | [x] There must be exactly one space between parameter type hint and parameter $a.
-----------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------
```

Running `phpcbf --standard=SlevomatCodingStandard --sniffs=SlevomatCodingStandard.TypeHints.ParameterTypeHintSpacing test.php` results in:

```php
<?php

function foo(
    #[A ]
    $a
) {
    return $a;
}
```

which isn't looking quite right.

If I add a parameter type:

```php
<?php

function foo(
    #[A]
    string $a
) {
    return $a;
}
```

Then the Sniff will not report an error.

see https://github.com/laminas/laminas-validator/pull/165#discussion_r1047818510 for a real-world use case.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/c51edb898bebd36aac70a190c6a41a7c056bb5b9

**Comment by TimWolla** [comment]
Thank *you* for the super fast turnaround time. I appreciate it.
